educator/src/Dscp/Educator/Web/Bot/Handlers.hs

```diff
@@ -52,19 +52,21 @@ addBotHandlers student StudentApiEndpoints{..} = botEndpoints
 
         , sMakeSubmission = \newSub -> do
             botProvideInitSetting student
-            res <- sMakeSubmission newSub
-
-            delayed $ requestToSignedSubmission newSub >>= botGradeSubmission
-
-            allAssigns <- sGetAssignments Nothing Nothing Nothing
-            botProvideUnlockedAssignments student res allAssigns
 
             -- cheat: on 3 submissions for the same assignment unlock all courses
```

Comment:
I do not understand how's that a fix. You've just moved a piece of code giving default assignments after the `when (length ...)` check.

Reply:
:heavy_plus_sign: , can you elaborate how this allows for multiple submissions for assignments?

Reply:
It does not (and afaiu we decided not to support multiple submissions on same assignment).
It just allows to make consequent failing submissions of the same assignment which would unlock advanced courses.

Reply:
>It just allows to make consequent failing submissions of the same assignment which would unlock advanced courses.

Can we just find a better way to unlock extra courses?